src/app/health/health-event-dialog.component.ts

```diff
@@ -46,4 +49,21 @@ export class HealthEventDialogComponent implements OnInit {
     this.router.navigate([ 'event', { id: this.data.user, eventId: event._id } ], { relativeTo: this.data.route });
   }
 
+  countdown() {
+    this.timer();
+    const intervalId = setInterval(() => {
+      this.timer(intervalId);
+    }, 1000);
+  }
+
+  timer(intervalId?) {
+    const time = (10000 - (new Date(Date.now()).getTime() - new Date(this.event.date).getTime())) / 1000;
+    if (time <= 0) {
+      clearInterval(intervalId);
```

Comment:
When the time hits 0 you should set `canUpdate` to `false`, also, so the button disappears.